File: TestBackEnd.sln/src/TestBackEnd.Domain/Entities/VehiclePosition.cs
namespace TestBackEnd.Domain.Entities
{
    public class VehiclePosition : Base
    {
        public double Latitude { get; private set; }
        
        public double Longitude { get; private set; }
        
        public long VehicleId { get; set; }

        public Vehicle Vehicle { get; set; }
        
        public VehiclePosition() { }

        public VehiclePosition(double latitude, double longitude, long vehicleId, Vehicle vehicle)
        {
            Latitude = latitude;
            Longitude = longitude;
            VehicleId = vehicleId;
            Vehicle = vehicle;
        }
    }
}

File: src/main/java/br/com/teste/java/testebackend/domain/VehiclePosition.java
package br.com.teste.java.testebackend.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Builder
public class VehiclePosition implements Serializable {

    private static final long serialVersionUID = -3656431259068389491L;

    @Id
    @GeneratedValue
    private Long id;
    private double latitude;
    private double longitude;
    private long vehicleId;
}
